### ISSUE TITLE
Refactor GHEventPayload inner classes into top level classes in a new namespace

### ISSUE THREAD

**Issue by bitwiseman** [hub4j/github-api#1058]
[GHEventPayload](https://github.com/hub4j/github-api/blob/master/src/main/java/org/kohsuke/github/GHEventPayload.java) is one monolithic class with a bunch of inner classes for payloads. We should look at refactoring these inner classes into their own files in a separate namespace.

**Comment by gsmet** [comment]
One important thing is that we need the name to be different from the base classes e.g. `GHPullRequest`.

Typically, I'm quite happy with how it looks when you use them such as in https://github.com/quarkusio/quarkus-bot/blob/main/src/main/java/io/quarkus/bot/AffectMilestone.java#L36 .

It's very readable, you know what you're manipulating and it's easy to avoid conflicts with the other classes.

**Comment by bitwiseman** [comment]
@gsmet 
I'm not sure I understand what you're suggesting, but it sounds reasonable.  The inner classes already have names different from the base, right?

**Comment by gsmet** [comment]
Yes, they have different names but I think we probably need to make it clear they are payloads or things could become quite confusing given you will use both the payloads and the API objects at the same time in most cases.
